src/quickgui/qgsquickpositionkit.cpp

```diff
@@ -218,3 +270,19 @@ bool QgsQuickPositionKit::simulated() const
 {
   return mIsSimulated;
 }
+
+void QgsQuickPositionKit::setMapSettings( QgsQuickMapSettings *mapSettings )
+{
+  if ( mMapSettings == mapSettings )
+    return;
+
+
+  if ( mMapSettings )
+  {
+    disconnect( mMapSettings, nullptr, this, nullptr );
```

Comment:
just ``disconnect(mMapSettings);` (or other way around i do not remember, check docs)